pkg/sql/schema_change_plan_node.go

```diff
@@ -57,7 +58,9 @@ func (p *planner) SchemaChange(ctx context.Context, stmt tree.Statement) (planNo
 		scs.stmts,
 	)
 	outputNodes, err := scbuild.Build(ctx, deps, scs.state, stmt)
-	if scerrors.HasNotImplemented(err) && mode == sessiondatapb.UseNewSchemaChangerOn {
+	if scerrors.HasNotImplemented(err) &&
+		(mode == sessiondatapb.UseNewSchemaChangerOn ||
+			mode == sessiondatapb.UseNewSchemaChangerUnsafe) {
```

Comment:
nit: consider testing for `mode != sessiondatapb.UseNewSchemaChangerUnsafeAlways` instead? It's explicitly the only case where we _don't_ fall back to the legacy schema changer.